Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -258,7 +258,10 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
           // It's not necessary to restate the postcondition itself if the user is already hovering it
           // however, nested postconditions should be displayed
           if (errorToken is BoogieRangeToken rangeToken && !hoveringPostcondition) {
-            deltaInformation += "  \n" + CouldProveOrNotPrefix + ideState.TextDocumentItem.Text.Substring(rangeToken.StartToken.pos,
+            var originalText =
+              ideState.Uri == rangeToken.Filename ? ideState.TextDocumentItem.Text :
+              File.ReadAllText(rangeToken.ActualFilename);
```

Comment:
I'd rather not add `File.ReadAllText` than being able to show the deltaInformation when the hover needs information from a file that's not `ideState.TextDocumentItem.Text`

I think the way this should work is: `deltaInformation += " 
  \n" + CouldProveOrNotPrefix + ((Program)program).FindNodeFromToken(rangeToken.StartToken).PrintOriginal()`

Reply:
Why would you look for a node in the tree when the rangeToken contains everything?

However, you have a point: since the tokens contain everything, we could just go from start to end and print tokens. I'll implement this instead.